runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkCachedArtifactPaths.java

```diff
@@ -0,0 +1,31 @@
+package org.apache.beam.runners.flink;
+
+/**
+ * Determines artifact path names within the
+ * {@link org.apache.flink.api.common.cache.DistributedCache}.
+ */
+public class FlinkCachedArtifactPaths {
+  private static final String DEFAULT_ARTIFACT_TOKEN = "default";
+
+  public static FlinkCachedArtifactPaths createDefault() {
+    return new FlinkCachedArtifactPaths(DEFAULT_ARTIFACT_TOKEN);
+  }
+
+  public static FlinkCachedArtifactPaths forToken(String artifactToken) {
+    return new FlinkCachedArtifactPaths(artifactToken);
+  }
+
+  private final String token;
+
+  private FlinkCachedArtifactPaths(String token) {
+    this.token = token;
+  }
+
+  public String getArtifactPath(String name) {
+    return String.format("ARTIFACT_%s_%s", token, name);
```

Comment:
If the artifact name can be an arbitrary string (which I assume is the case), we'll need to encode this safely. Base64Url should be good.

Reply:
As we discussed offline, it looks like the ULR does not currently do this escaping for incoming artifacts. We'll need to fix that in the service implementation before it does anything here. Please update https://issues.apache.org/jira/browse/BEAM-3672 with a note about this and insert a TODO here, linking to that bug.

Reply:
Discussed offline.

While this is a legitimate objection, it is part of a larger story about how we handle artifact staging.  The naivete of the current implementation is reflected in additional upstream sources (specifically those in `/runners/local-artifact-service-java`.  We decided that the best solution was to hold off on migrating to a safe encoding for a subsequent PR.